src/lib/Repository/UiTranslationService.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\HybridPlatformUi\Repository;
+
+use eZ\Publish\API\Repository\LanguageService;
+use eZ\Publish\API\Repository\Values\Content\VersionInfo;
+
+/**
+ * Service for loading translations.
+ */
+class UiTranslationService
```

Comment:
What's the reason for being skeptical? I don't think it would make it too complex but it would mean that it no longer has a single responsibility as it would be for loading users and loading translations